src/test/incremental/hashes/trait_defs.rs

```diff
@@ -303,7 +303,7 @@ trait TraitChangeModeSelfOwnToMut: Sized {
 #[cfg(not(cfail1))]
 #[rustc_clean(label="Hir", cfg="cfail2")]
 #[rustc_clean(label="Hir", cfg="cfail3")]
-#[rustc_metadata_dirty(cfg="cfail2")]
+#[rustc_metadata_clean(cfg="cfail2")]
```

Comment:
Can we make an ICH test targeting exactly this change? Seems easily do-able. In particular, if you have something like `fn foo(x: u32) { }` and you change it to `fn foo(y: u32) { }`, the hash of the trait should not change (just the body). Right?

That said, if you don't provide a default, based on my read of the code above, then the fn names would still be included in the hash... which doesn't seem exactly *right*. We could I guess just ignore them manually in the hasher code.